src/auth/negotiate/kerberos/negotiate_external_acl

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#
+# This external  acl helper checks the Kerberos group information without
+# continues ldap queries by leveraging the kerbors ticket group information
+# i.e. PAC field
+#
+# It requires only during startup aldap  query to get the SID for the group name.
+#
+# Squid config lines:
+# auth_param negotiate program /opt/squid/sbin/negotiate_wrapper_auth -d --ntlm /usr/bin/ntlm_auth --helper-protocol=squid-2.5-ntlmssp --domain SAMBA.HOME --kerberos /opt/squid/sbin/negotiate_kerberos_auth -d -s GSS_C_NO_NAME -k /etc/squid/squid.keytab -t none
+# external_acl_type test_acl ipv4 %LOGIN %note{group} /opt/squid/sbin/negotiate_external_acl
+# acl squid_allow external test_acl
+#
+# GET SID for Group
+#
+# Please change sample values with real valuse
+#
+if [ -z "$KRB5_KTNAME" ]; then
+   export KTNAME="/etc/krb5.keytab"
+else
+   export KTNAME=$KRB5_KTNAME
+fi
+export PRINCIPAL="HTTP/opensus42.suse.home"
+export DC="dc1.samba.home"
+export KRB5CCNAME="/tmp/squid_krb5cc"
+export GROUPNAME="SQUID_ALLOW"
+export BASEDN="DC=samba,DC=home"
+#
+#
+#
+
+kinit -kt $KTNAME $PRINCIPAL
+SID=`ldapsearch -LLL -Ygssapi -H ldap://$DC:389 -s sub -b "BASEDN" "(CN=$GROUPNAME)" objectsid 2>&1 | awk '{ if ( $0 ~/^object/ ) print $2}'`
+
+(>&2 echo "`date +"%Y/%m/%d %H:%M:%S"`| negotiate_external_acl: SID=$SID")
+
+#
+# Loop over input
+#
+while [ 1 == 1 ] ; do
```

Comment:
There does not seem to be any exit/break to leave this loop. The script does need to terminate cleanly if Squid closes the I/O socket this script sees as stdin.